support-frontend/assets/pages/contributions-landing/components/AmazonPay/useAmazonPayObjects.ts

```diff
@@ -0,0 +1,41 @@
+import { useEffect } from 'preact/hooks';
+import { useState } from 'react';
+import { setupAmazonPay } from 'helpers/forms/paymentIntegrations/amazonPay/setup';
+import type {
+	AmazonLoginObject,
+	AmazonPaymentsObject,
+} from 'helpers/forms/paymentIntegrations/amazonPay/types';
+import type { CountryGroupId } from 'helpers/internationalisation/countryGroup';
+
+interface AmazonPayObjects {
+	loginObject?: AmazonLoginObject;
+	paymentsObject?: AmazonPaymentsObject;
+}
+
+export function useAmazonPayObjects(
+	hasSelectedAmazonPay: boolean,
+	countryGroupId: CountryGroupId,
+	isTestUser: boolean,
+): AmazonPayObjects {
+	const [loginObject, setLoginObject] = useState<
+		AmazonLoginObject | undefined
+	>();
+
+	const [paymentsObject, setPaymentsObject] = useState<
+		AmazonPaymentsObject | undefined
+	>();
+
+	useEffect(() => {
+		const hasLoaded = loginObject !== undefined || paymentsObject !== undefined;
+
+		if (hasSelectedAmazonPay && !hasLoaded)
+			setupAmazonPay(
+				countryGroupId,
+				setLoginObject,
+				setPaymentsObject,
+				isTestUser,
+			);
+	}, [hasSelectedAmazonPay]);
```

Comment:
I wasn't quite sure what the best practice was to do here. Technically, `loginObject` and `paymentsObject` should be in the dependencies array, as they are referenced in side of the effect. However, I explicitly don't want the transition from `undefined` to defined to retrigger the effect (the `if` statement will prevent the `setupAmazonPay` running again, but still feels slightly watsted to trigger the effect again).